pkg/cmd/util/clientcmd/factory.go

```diff
@@ -201,7 +201,11 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			if !ok {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
-			buildsForBCSelector := labels.SelectorFromSet(map[string]string{buildapi.DeprecatedBuildConfigLabel: t.Name})
+			buildConfigLabels := labels.Set{
+				buildapi.BuildConfigLabel: t.Name,
```

Comment:
realistically, probably never since we have no way to guarantee old content(which lacks the new label) will ever be removed from etcd.